src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = (T[]) new Object[currentSize];
 
+    public void push(T value) {
+        if (size >= currentSize) {
+            table = grow();
+        }
+        table[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return table[size - 1];
     }
 
     public T pop() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        T value = peek();
+        table[size - 1] = null;
+        size--;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private T[] grow() {
+        double newLength = table.length * 1.5;
+        currentSize = (int) newLength;
```

Comment:
The growth factor of 1.5 is a good choice for amortized growth, but you should consider defining it as a constant to avoid magic numbers in your code and to make it clear what the value represents.

Reply:
ok, done